lib/array_intersection.rb

```diff
@@ -1,4 +1,17 @@
 def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
+  num_hash = {}
+  intersection = []
 
-end
\ No newline at end of file
+  list1.each do |num|
+    num_hash[num] = nil
```

Comment:
Generally hashes return `nil` if a key isn't in the hash, so it makes more sense to use another value than `nil`.

```suggestion
    num_hash[num] = true
```